src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -42,9 +43,34 @@
 #include "gc/shenandoah/shenandoahVMOperations.hpp"
 #include "gc/shenandoah/shenandoahWorkGroup.hpp"
 #include "gc/shenandoah/shenandoahWorkerPolicy.hpp"
+#include "memory/allocation.hpp"
 #include "prims/jvmtiTagMap.hpp"
+#include "runtime/vmThread.hpp"
 #include "utilities/events.hpp"
 
+// Breakpoint support
+class ShenandoahConcurrentGCScope : public StackObj {
```

Comment:
Let's call these `ShenandoahGCBreakpointScope` and `ShenandoahMarkBreakpointScope`?

Reply:
Done